### DISCUSSION TITLE
delete in chunks in postgresql

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#3455] [Ideas]
Apparently, postgresql does not support deleting in chunks directly. Thus lingtodb's SQL of: Delete(q.Take(n)) won't work.

Why don't we change the SQL a little bit only for PostgreSQL? It's what the LinqToDB.PostgreSQL is for, right?
We can easily change it to:

delete table where ctid in select ctid from q.take(n);

**Comment by MaceWindu** [comment]
PR is welcome
